servers/zms/src/main/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnection.java

```diff
@@ -3355,6 +3363,89 @@ public DomainRoleMembers listDomainRoleMembers(String domainName) {
         return domainRoleMembers;
     }
 
+
+    @Override
+    public boolean confirmRoleMember(String domainName, String roleName, RoleMember roleMember,
+                                    String admin, String auditRef) {
+
+        final String caller = "confirmRoleMember";
+
+        String principal = roleMember.getMemberName();
+        int domainId = getDomainId(domainName);
+        if (domainId == 0) {
+            throw notFoundError(caller, ZMSConsts.OBJECT_DOMAIN, domainName);
+        }
+        int roleId = getRoleId(domainId, roleName);
+        if (roleId == 0) {
+            throw notFoundError(caller, ZMSConsts.OBJECT_ROLE, ZMSUtils.roleResourceName(domainName, roleName));
+        }
+        if (!validatePrincipalDomain(principal)) {
```

Comment:
if we're confirming a role member, there is no need to validate domain nor insert the principal if one doesn't exist since those are checks done by the insert call. so 3382-3384 and 3387 should not be necessary